src/node_crypto.cc

```diff
@@ -2893,6 +2898,7 @@ void CipherBase::SetAuthTag(const FunctionCallbackInfo<Value>& args) {
 bool CipherBase::SetAAD(const char* data, unsigned int len, int plaintext_len) {
   if (!ctx_ || !IsAuthenticatedMode())
     return false;
+  ClearErrorOnReturn clear_error_on_return;
```

Comment:
This should ideally be `MarkPopErrorOnReturn`.  `ClearErrorOnReturn` is a blunt hammer and might swallow genuine errors.

Reply:
Point taken, I can change it. In practice, the error stack should be empty when the JS layer calls into the C++ layer though, right?

Reply:
Yes, except for the following:

1. This could mess up C++ -> C++ calls when the caller also manipulates the error stack.
2. I know of at least one add-on that leaves openssl errors on the stack for some time.